### DISCUSSION TITLE
Including tos-optee and related packages to rootfs by default

### DISCUSSION THREAD

**Discussion by indresharma** [OE4T/meta-tegra#1346] [Q&A]
Hi All,

Packages related to tos-optee like optee-os and arm-trusted-formware are not included in the rootfs image by default
I am using meta-tegra to build 'core-minimal-image' on 'jetson-agx-orin-devkit' machine. Yocto release is kirkstone.
I do see a tos-jetson-agx-orin-devkit-35.3.1-r0.img file in deploy/images folder.

However when I add these packages using IMAGE_INSTALL:append = "optee-os arm-trusted-firmware" these are included in rootfs.

Is there any other way to include all tos-optee related packages by default to rootfs image?

It would be a great help if anyone could point out related documentation.

Below is rootfs manifest generated:
```
base-files jetson_agx_orin_devkit 3.0.14-r89
base-passwd armv8a 3.5.29-r0
busybox armv8a_tegra 1.35.0-r0
busybox-hwclock armv8a_tegra 1.35.0-r0
busybox-syslog armv8a_tegra 1.35.0-r0
busybox-udhcpc armv8a_tegra 1.35.0-r0
eudev armv8a 3.2.10-r0
init-ifupdown jetson_agx_orin_devkit 1.0-r7
init-system-helpers-service armv8a 1.62-r0
initscripts armv8a 1.0-r155
initscripts-functions armv8a 1.0-r155
ldconfig armv8a 2.35-r0
libblkid1 armv8a 2.37.4-r0
libc6 armv8a 2.35-r0
libkmod2 armv8a 29-r0
liblzma5 armv8a 5.2.6-r0
libz1 armv8a 1.2.11-r0
modutils-initscripts armv8a 1.0-r7
netbase all 1:6.3-r0
packagegroup-core-boot jetson_agx_orin_devkit 1.0-r17
sysvinit armv8a 3.01-r0
sysvinit-inittab jetson_agx_orin_devkit 2.88dsf-r10
sysvinit-pidof armv8a 3.01-r0
tegra-firmware armv8a_tegra234 35.3.1-20230319081403-r0
tegra-firmware-tegra234 armv8a_tegra234 35.3.1-20230319081403-r0
tegra-firmware-vic armv8a_tegra234 35.3.1-20230319081403-r0
tegra-firmware-xusb armv8a_tegra234 35.3.1-20230319081403-r0
update-alternatives-opkg armv8a 0.5.0-r0
update-rc.d all 0.8-r0
```

**Comment by ichergui** [comment]
Hey @indresharma 
optee-os and ATF shouldn't be installed into the rootfs since they live in their own partition. 
optee-client, optee-test and Nvidia optee-samples are shipped into the rootfs.

**Reply by indresharma** [reply]
Hi @ichergui 

Thanks for your quick response. 

I don't see optee-client, optee-test and Nvidia optee-samples either in the rootfs manifest. 
Is there a config I need to define to include these in rootfs manifest ?

The manifest is shared above.

**Reply by madisongh** [reply]
`core-image-minimal` does not include those packages (`optee-nvsamples` and `optee-test`) by default -- the minimal image includes almost nothing by default, hence the name.  What I think @ichergui meant was that you _can_ configure your build to include them in the rootfs.  Adding `optee-os` and `arm-trusted-firmware` to the rootfs does nothing, since those are part of the boot firmware.

We _do_ include the samples and tests in the `demo-image-xxxx` images in our [demo distro](https://github.com/OE4T/tegra-demo-distro).  It might be easier to use that for testing, since the demo images include other packages you're likely to need on the Jetson hardware.
